File: LeetCode/976. Largest Perimeter Triangle/Solution.py
from typing import List

class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        # Sort ascending so we can check triples from the largest side
        nums.sort()
        # Trarvese from the end; check consecutive triplses (i-2, i-1, i)
        for i in range(len(nums) - 1, 1, -1):
            a, b, c = nums[i - 2], nums[i - 1], nums[i]
            if a + b > c:
                return a + b + c
        
        return 0

File: LeetCode/976. Largest Perimeter Triangle/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
    public:
    int largestPerimeter(vector<int>& nums) {
        sort(nums.begin(), nums.end())
        for (int i = (int)nums.size() - 1; i >=2; --i) {
            int a = nums[i - 2], b = nums[i - 1], c = nums[i];
                // For sorted size a <= b <= c, it's sufficient to check a + b > c
                if ((long long)a + b > c) {
                    return a + b + c;
                }

        }

        return 0; // No case found
    }
}